programs/stake/src/stake_state.rs

```diff
@@ -655,62 +647,59 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
                         (remaining_stake_delta, remaining_stake_delta)
                     } else {
                         // Otherwise, the new split stake should reflect the entire split
-                        // requested, less any lamports needed to cover the split_rent_exempt_reserve
+                        // requested, less any lamports needed to cover the split_rent_exempt_reserve.
+                        let split_lamports = split.lamports()?;
+                        let split_rent_exempt_reserve_deficit = validated_split_info
+                            .destination_rent_exempt_reserve
+                            .saturating_sub(split_lamports);
+                        let minimum_split_amount = MINIMUM_STAKE_DELEGATION
+                            .saturating_add(split_rent_exempt_reserve_deficit);
+                        if lamports < minimum_split_amount {
+                            return Err(InstructionError::InsufficientFunds);
+                        }
```

Comment:
well, i think this logic is desired to be encapsulated inside `.validate_split_amount()` while being enclosed inside the new `if source_remaining_balance != 0 {...}`.

That's because this is actually another new validation code, which we're trying to refactor all of 'em. :)

Reply:
I was forgetting that jon is suggesting a similar: https://github.com/solana-labs/solana/pull/22663/files#r818061753 and that the replaced logic is shared for `StakeState::Stake` and `StakeState::Initialized(meta)`....

I'd suggest passing a flag like `is_stake_active` to `validate_split_amount()`...

Reply:
Ok, done!

Instead of passing a bool to `validate_split_amount()`, I've instead gone with `Option<&Stake>`. It functions similar to a bool, but now there's additional type safety to ensure there's actually a staked account being split.